src/modules/rating/rating.ts

```diff
@@ -0,0 +1,28 @@
+import { ObjectId } from 'mongodb';
+import { useCollection } from 'db';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { RatingForm } from 'prytaneum-typings';
+
+import createHttpError from 'http-errors';
+
+const addRating = async (
+    rating: RatingForm,
+    townhallId: string,
+    userId?: string
+): Promise<void> => {
+    const filter = { _id: new ObjectId(townhallId) };
+    const update = userId
```

Comment:
This needs a bit of modification.

The interface in `prytaneum-typings` is as follows: 
```ts
export interface Rating<T extends string | ObjectId = string> {
    _id: T;
    ratings: Set<{
        userId: T;
        values: Array<{ question: string; value: number | null }>;
        feedback: string;
    }>;
}
```

The interface is requiring the `userId` be defined for every single element.  

Lets change the `ratings` field to the following:
```ts
{ 
    ratings: { userId?: T, values: Record<string, string | null> }[]
}
```

This is advantageous because of the following:
Using an array makes the `values` field a nested array and harder to query with no real advantages  -- we never leverage the array type to our advantage.  It's true that we don't always know what keys will be, since that is configured by the user. However, that is why we're defining the `values` type as a `Record` with any `string` key that will have a `number | null` value.

If you think about it, an array is just an object whose keys are index values. Whereas an object is an array whose keys are strings.  You can also think about this the inverse way; an object is just an array whose index values are strings. Each comes with its own advantages and disadvantages.

Reply:
Okay, I updated this so it will use records. This also made the front end a bit easier to work with once I updated it. Do you know if the Joi validation treats the Record type as just an object? 

Also, I updated the RatingForm to just include the optional userId, that way we don't even need to use a query param for it.  Or would you rather me put it back to a query param, not sure if there might be some reason to prefer using params over optional body params.

Reply:
Yes, the record type shown above is equivalent to the following: 
```ts
interface SomeType {
    [index: string]: number | null;
}
```
The record type is just more succinct for this particular case.

Also, bear in mind that the type Record is a compile/transpile time check.  Whereas Joi is a runtime check.  So Joi actually has no idea what a Record is, only typescript does.  To answer your question, yes, Joi can handle that type (technically almost everything in js is an object, including functions).  

Query params are only preferable if we care about pre-flight checks.  We sorta do in that if we can avoid them, then we probably should.  So, if it doesn't make life too much harder, then yeah I would make it a query param. 

I haven't looked closely yet at the API design itself, so if you want to hold off on changing that until I look at that more closely, then that's fine too.  My first pass was just looking for things that jumped out to me.

Reply:
It's really not much trouble to change it back to query params, but perhaps when you look at the overall design you will change your mind so I'll hold off on changing it for now.